cloud_chemistry/miam/README-example.cpp

```diff
@@ -0,0 +1,159 @@
+     .name_ = "SMALL_DROP",
+     .shape_ = DistributionType::SingleMomentMode, // tracks total mass in state; fixed radius; number calculated
+     .phases_ = { aqueous_phase }
+  }};
+  auto large_drop = AerosolDistribution{{
+     .name_ = "LARGE_DROP",
+     .shape_ = DistributionType::SingleMomentMode, // tracks total mass in state; fixed radius; number calculated
+     .phases_ = { aqueous_phase }
+  }};
+  AerosolModel cloud{{
+    .name_ = "CLOUD",
+    .distributions_ = { small_drop, large_drop }
+  }};
+
+  auto aitken = AerosolDistribution{{
+    .name_ = "AITKEN",
+    .shape_ = DistributionType::TwoMomentMode, // tracks total mass and number concentration in state; radius calculated
+    .phases_ = { organic_phase }
+  }};
+  auto accumulation = AerosolDistribution{{
+    .name_ = "ACCUMULATION",
+    .shape_ = DistributionType::TwoMomentMode, // tracks total mass and number concentration in state; radius calculated
+    .phases_ = { aqueous_phase, organic_phase }
+  }};
+  AerosolModel aerosol{{
+    .name_ = "AEROSOL",
+    .distributions_ = { aitken, accumulation }
+  }};
+  
+  // State array should contain
+  //  1) GAS.CO2
+  //  2) CLOUD.SMALL_DROP.AQUEOUS.CO2
+  //  3) CLOUD.SMALL_DROP.AQUEOUS.H2O
+  //  4) CLOUD.SMALL_DROP.AQUEOUS.OH-
+  //  5) CLOUD.SMALL_DROP.AQUEOUS.H+
+  //  6) CLOUD.SMALL_DROP.AQUEOUS.HCO3-
+  //  7) CLOUD.SMALL_DROP.AQUEOUS.CO32-
+  //  8) CLOUD.LARGE_DROP.AQUEOUS.CO2
+  //  9) CLOUD.LARGE_DROP.AQUEOUS.H2O
+  // 10) CLOUD.LARGE_DROP.AQUEOUS.OH-
+  // 12) CLOUD.LARGE_DROP.AQUEOUS.H+
+  // 12) CLOUD.LARGE_DROP.AQUEOUS.HCO3-
+  // 13) CLOUD.LARGE_DROP.AQUEOUS.CO32-
+  // 14) AEROSOL.AIKEN.ORGANIC.CO2
+  // 15) AEROSOL.AIKEN.ORGANIC.HEXANE
+  // 16) AEROSOL.ACCUMULATION.AQUEOUS.CO2
+  // 17) AEROSOL.ACCUMULATION.AQUEOUS.H2O
+  // 18) AEROSOL.ACCUMULATION.AQUEOUS.OH-
+  // 19) AEROSOL.ACCUMULATION.AQUEOUS.H+
+  // 20) AEROSOL.ACCUMULATION.AQUEOUS.HCO3-
+  // 21) AEROSOL.ACCUMULATION.AQUEOUS.CO32-
+  // 22) AEROSOL.ACCUMULATION.ORGANIC.CO2
+  // 23) AEROSOL.ACCUMULATION.ORGANIC.HEXANE
+  
+
+  Process co2_photo = ChemicalReactionBuilder()
+                   .SetReactants({ co2 })
+                   .SetRateConstant(ArrheniusRateConstant({ .A_ = 1.0e-3 }))
+                   .SetPhase(gas_phase);
+
+  // Build an Effective Henry's Law PhaseTransferProcess
+  // (will be for a diprotic acid: CO2(g) <-> H2CO3(aq) <-> HCO3- + H+ <-> CO32- + 2H+)
+  // K_a1 = first acid dissociation constant
+  // K_a2 = second acid dissociation constant
+  // K_H = A * exp(C / T) = Henry's Law constant
+  auto hlc_co2 = HenrysLawCoefficient{ A_ = 32.4, C_ = -3.2e-3, K_a1_ = 1.0e-5, K_a2_ = 2.0e-5 };
+  Process co2_phase_transfer = PhaseTransferProcessBuilder()
+                    .SetGasSpecies( gas_phase, co2 )
+                    .SetCondensedSpecies( aqueous_phase, Yield( 2, hp ), co32m )
+                    .SetSolvent( h2o )
+                    .SetTransferCoefficient( hlc_co2 )
+                    .Build();
```

Comment:
Do we expect the setter for gas or condensed species to have more than one species in a phase transfer reaction? If so, should it take a vector of species, or a vector of Yield?
```suggestion
  auto hlc_co2 = HenrysLawCoefficient{ A_ = 32.4, C_ = -3.2e-3, K_a1_ = 1.0e-5, K_a2_ = 2.0e-5 };
  Process co2_phase_transfer = PhaseTransferProcessBuilder()
                    .SetGasSpecies( gas_phase, { co2 } )
                    .SetCondensedSpecies( aqueous_phase, { Yield( hp, 2 ), Yield(co32m) } )
                    .SetSolvent( aqueous_phase, h2o )
                    .SetTransferCoefficient( hlc_co2 )
                    .Build();
```

Reply:
~~Also, what do you think about storing a pointer to the phase in the `Process` class? I'm still unsure whether we actually need to store the phase to access the diffusion coefficient. I believe that's a species property, which we should be able to retrieve directly from the species~~

Reply:
I like that. There will only ever be one gas-phase species, but if we support regular and effective Henry's Law partitioning, there could be multiple condensed-phase species. I'll update the `SetCondensedSpecies()` to take a vector of species.